server/auth-strategies/ldap.js

```diff
@@ -28,26 +28,59 @@ function enableLdap(config) {
           bindCredentials:
             config.get('ldapPassword') || config.get('ldapPassword_d'),
           searchFilter: config.get('ldapSearchFilter'),
+          groupSearchBase: config.get('ldapBaseDN'),
+          groupSearchFilter: '(cn={{dn}})',
         },
       },
       async function passportLdapStrategyHandler(req, profile, done) {
         try {
           const { models } = req;
-          const mail = profile.mail.toLowerCase();
-          const user = await models.users.findOneByEmail(mail);
-          if (!user) {
+          const email = profile.mail.toLowerCase();
+          const uid = profile.uid.toLowerCase();
+          const admin_group = config.get('sqlpadLadpAdminGroupDn');
+          const editor_group = config.get('sqlpadLadpEditorGroupDn');
+
+          // get all groups that user belongs to, not sure how to substitue memberOf from env var
+          const groups = profile.memberOf;
+          // match the groups which are predefined, refer to configuration
+          if (groups.includes(admin_group)) {
+            appLog.debug(`${uid} successfully logged in with role admin`);
+            global.role = 'admin';
+          } else if (groups.includes(editor_group)) {
+            appLog.debug(`${uid} successfully logged in with role editor`);
+            global.role = 'editor';
+          }
```

Comment:
Instead of using `global.role`, could we use a local `let role;` instead? 

`let` allows changing the value assignment in JS. For example you can do

```js
let role;
if (groups.includes(admin_group)) {
  role = 'admin';
} else if (groups.includes(editor_group)) {
  role = 'editor';
}
```

You may get a lint warning later, since the variable name is the same name as the key. This is fine but there's a shorthand you can use. 

This:
```js
const newUser = await models.users.update(user.id, {
  role: role,
});
```

can become this (as long as the local variable `role` is in scope)

```js
const newUser = await models.users.update(user.id, {
  role,
});
```

Reply:
Thank you for the feedback @rickbergfalk , I'll make the changes , will need to test it though :) will try to find some time this week.